app/src/main/java/org/simple/clinic/user/UserSession.kt

```diff
@@ -108,27 +111,16 @@ class UserSession @Inject constructor(
   }
 
   fun findExistingUser(phoneNumber: String): Single<FindUserResult> {
-    Timber.i("Finding user with phone number")
-    return registrationApi.findUser(phoneNumber)
-        .map { user -> FindUserResult.Found(user) as FindUserResult }
-        .onErrorReturn { e ->
-          when {
-            e is IOException -> FindUserResult.NetworkError
-            e is HttpException && e.code() == 404 -> FindUserResult.NotFound
-            else -> {
-              Timber.e(e)
-              FindUserResult.UnexpectedError
-            }
-          }
-        }
+    return findUserWithPhoneNumber.find(phoneNumber)
   }
 
   fun refreshLoggedInUser(): Completable {
     return requireLoggedInUser()
         .firstOrError()
         .doOnSuccess { Timber.i("Refreshing logged-in user") }
         .flatMapCompletable { loggedInUser ->
-          registrationApi.findUser(loggedInUser.phoneNumber)
+          findUserWithPhoneNumber.find(loggedInUser.phoneNumber)
```

Comment:
Hrm… here we are finding a user by his phone number. Should this be,

`userFinder.findByPhoneNumber(String)` / `userApi.findByPhoneNumber(String)`?

Reply:
See earlier comment about calling the class `UserLookup`?